src/SFML/Network/IpAddress.cpp

```diff
@@ -163,10 +176,13 @@ std::optional<IpAddress> IpAddress::getPublicAddress(Time timeout)
     Http                 server("www.sfml-dev.org");
     const Http::Request  request("/ip-provider.php", Http::Request::Method::Get);
     const Http::Response page = server.sendRequest(request, timeout);
-    if (page.getStatus() == Http::Response::Status::Ok)
+
+    const Http::Response::Status status = page.getStatus();
+
+    if (status == Http::Response::Status::Ok)
         return IpAddress::resolve(page.getBody());
 
-    // Something failed: return an invalid address
+    err() << "Failed to retrieve public address (HTTP response status" << static_cast<int>(status) << ")" << std::endl;
```

Comment:
This might be rather confusing/leaking implementation. Why does my public IP resolving failed with an HTTP error code?

Reply:
We are already clearly documenting that `getPublicAddress` depends on an external server:

```cpp
    /// ...
    /// this function depends on the network connection and on a distant
    /// server
    /// ...
```

I think knowing why access to the server failed is useful in debugging unexpected `getPublicAddress` failures. I will tweak the message a bit to make it much clear.